homeassistant/components/upnp/__init__.py

```diff
@@ -121,6 +121,9 @@ async def device_discovered(
         cancel_discovered_callback()
 
     # Create device.
+    if TYPE_CHECKING:
```

Comment:
Should these assertions always be enabled? If we believe they're always true then there's no harm removing the `if TYPE_CHECKING`, and if they're not always true then we're lying to the type checker.

Reply:
I have adjusted this